06-building-decays.md

```diff
@@ -0,0 +1,142 @@
+> A `Selection` requires *all* its `RequiredSelections` to pass.
+>
+> There are several types of selections, namely
+>
+> - `Selection`
+> - `MergedSelection`, which is used to join the output of several `Selection` objects in a single TES location.
+> - `AutomaticData` (also known as `DataOnDemand`), which builds objects from their TES location using a preconfigured map of (location, algorithm) pairs.
+
+> ## Selection sequences {.callout}
+> The `SelectionSequence` takes a `Selection` object, resolves its `Selection` requirements, and builds a flat, chained and ordered list of `Selections`. It then exports (via the `selection` method) a self-contained `GaudiSequencer` with all the algorithm configurables necessary to run the selection.
+> It also makes the output locations of the data written by the head of the selection chain available via the `outputLocations` method.
+
+To get our input particles we have two equivalent options:
+
+ - Use the `DataOnDemand` class with the `Phys/StdAllNoPIDsPions/Particles` and `Phys/StdAllLooseKaons/Particles` as inputs for pions and kaons, respectively.
+
+```python
+from PhysSelPython.Wrappers import DataOnDemand
+Pions = DataOnDemand('Phys/StdAllNoPIDsPions/Particles')
+Kaons = DataOnDemand('Phys/StdAllLooseKaons/Particles')
+```
+
+ - Load the equivalent particle creation algorithms from the `CommonParticles` package:
+
+```python
+from CommonParticles.StdAllNoPIDsPions import StdAllNoPIDsPions as Pions
+from CommonParticles.StdAllLooseKaons import StdAllLooseKaons as Kaons
+```
+
+The `CommonParticles` package, which you can find [here](https://svnweb.cern.ch/trac/lhcb/browser/Stripping/trunk/Phys/CommonParticles/python/CommonParticles), allows to access premade particles with reasonable selections and gives easy access to common particles used in selections.
+If you access the code for `StdAllNoPIDsPions`, you can see that in it the algorithm for creating the particles is instantiated and the map of the `DataOnDemand` service is updated via the `updateDoD` function.
+
+Once we have the input pions and kaons, we can combine them to build a D0 by means of the `CombineParticles` algorithm.
+This algorithm performs the combinatorics for us according to a given decay descriptor and puts the resulting particle in the TES, allowing also to apply some cuts on them:
+
+ - `DaughtersCuts` is a dictionary that maps each child particle we want to cut on with a selection LoKi particle functor (or combination of functors) which results in a `bool`. Optionally, one can specify also a `Preambulo` property that allows us to make imports, preprocess functors, etc (more on this in the [LoKi functors](https://lhcb.github.io/first-analysis-steps/06-loki-functors.html) lesson). For example:
```

Comment:
"we want to cut on with a" -> "type to a LoKi particle functor that determines if that particular particle passes" not perfect :-/